test/minify.test.js

```diff
@@ -0,0 +1,170 @@
+          if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
+            expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
+          }
+        }
+      });
+  });
+
+  it('matches snapshot for `uglify-es` minifier', () => {
+    const compiler = createCompiler({
+      entry: `${__dirname}/fixtures/minify/es6.js`,
+      output: {
+        path: `${__dirname}/dist-uglify-es`,
+        filename: '[name].js',
+        chunkFilename: '[id].[name].js',
+      },
+    });
+
+    new UglifyJsPlugin({
+      minify(file) {
+        return uglifyEs.minify(file, {
+          mangle: {
+            reserved: ['baz'],
+          },
+        });
+      },
+    }).apply(compiler);
+
+    return compile(compiler)
+      .then((stats) => {
+        const errors = stats.compilation.errors.map(cleanErrorStack);
+        const warnings = stats.compilation.warnings.map(cleanErrorStack);
+
+        expect(errors).toMatchSnapshot('errors');
+        expect(warnings).toMatchSnapshot('warnings');
+
+        for (const file in stats.compilation.assets) {
+          if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
+            expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
+          }
+        }
+      });
+  });
+
+  it('matches snapshot for `terser` minifier', () => {
+    const compiler = createCompiler({
+      entry: `${__dirname}/fixtures/minify/es6.js`,
+      output: {
+        path: `${__dirname}/dist-terser`,
+        filename: '[name].js',
+        chunkFilename: '[id].[name].js',
+      },
+    });
+
+    new UglifyJsPlugin({
+      minify(file) {
+        return terser.minify(file, {
+          mangle: {
+            reserved: ['baz'],
+          },
+        });
+      },
+    }).apply(compiler);
+
+    return compile(compiler)
+      .then((stats) => {
+        const errors = stats.compilation.errors.map(cleanErrorStack);
+        const warnings = stats.compilation.warnings.map(cleanErrorStack);
+
+        expect(errors).toMatchSnapshot('errors');
+        expect(warnings).toMatchSnapshot('warnings');
+
+        for (const file in stats.compilation.assets) {
+          if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
+            expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
+          }
+        }
+      });
+  });
+
+  it('matches snapshot for `terser` minifier and `sourceMap: true`', () => {
+    const compiler = createCompiler({
+      entry: `${__dirname}/fixtures/minify/es6.js`,
+      output: {
+        path: `${__dirname}/dist-terser`,
+        filename: '[name].js',
+        chunkFilename: '[id].[name].js',
+      },
+    });
+
+    function removeAbsoluteSourceMapSources(source) {
+      if (source.map && source.map.sources) {
+        // eslint-disable-next-line no-param-reassign
+        source.map.sources = source.map.sources.map(sourceFromMap => path.relative(process.cwd(), sourceFromMap));
```

Comment:
any way to copy and reassign instead of directly mutating here? Perhaps extracting into a variable and reassigning would be better https://stackoverflow.com/questions/34716651/js-array-prototype-map-happens-to-be-mutable

Reply:
@ev1stensberg why? It is increase lines of code and decrease perf, but we just need remove absolute path. I don't think it is critical here.